### DISCUSSION TITLE
Save replay when failed

### DISCUSSION THREAD

**Discussion by cdwcgt** [ppy/osu#18770] [Ideas] [answered]
It's a feature on stable, there's no reason why it shouldn't be implemented on lazer

I tried to write an implementation and design like this
![image](https://user-images.githubusercontent.com/34775378/174485262-4722a8d4-40ec-4262-a483-ba324136652a.png)

https://user-images.githubusercontent.com/34775378/174485537-1ef43b71-ae9d-49ba-881c-15970a417e23.mp4

**Comment by bdach** [comment, marked as answer]
[please use the search box](https://github.com/ppy/osu/issues/8721)

**Reply by cdwcgt** [reply]
I tried to use `replay failed` to search but it nothing...

and I want to know can I open a pull request for it

https://github.com/cdwcgt/osu/tree/Save-Score-Failed

**Reply by bdach** [reply]
depends. the main issue is going to be the rank of the saved score. it should be a separate rank for failures i think.

this blocked [the last attempt to implement this](https://github.com/ppy/osu/pull/10339#issuecomment-702923810). which was linked in the issue. i would strongly recommend reading up on that pull request.

**Reply by cdwcgt** [reply]
ok, thanks for reply

**Comment by peppy** [comment]
The implementation looks fine how you have it. Feel free to PR it once you're ready.
